lib/active_merchant/billing/gateways/pay_trace.rb

```diff
@@ -0,0 +1,259 @@
+
+          response = commit('transactions/sale/keyed', post)
+          check_token_response(response, 'transactions/sale/keyed', post, options)
+        end
+      end
+
+      def authorize(money, payment_or_customerid, options = {})
+        post = {}
+        add_invoice(post, money, options)
+        if payment_or_customerid.class == String
+          post[:customer_id] = payment_or_customerid
+
+          response = commit('transactions/authorization/by_customer', post)
+          check_token_response(response, 'transactions/authorization/by_customer', post, options)
+        else
+          add_payment(post, payment_or_customerid)
+          add_address(post, payment_or_customerid, options)
+          add_customer_data(post, options)
+
+          response = commit('transactions/authorization/keyed', post)
+          check_token_response(response, 'transactions/authorization/keyed', post, options)
+        end
+      end
+
+      def capture(authorization, options = {})
+        post = {}
+        post[:amount] = options[:amount] if options[:amount]
+        post[:transaction_id] = authorization
+        response = commit('transactions/authorization/capture', post)
+        check_token_response(response, 'transactions/authorization/capture', post, options)
+      end
+
+      def refund(authorization, options = {})
+        # currently only support full and partial refunds of settled transactions via a transaction ID
+        post = {}
+        post[:amount] = options[:amount] if options[:amount]
+        post[:transaction_id] = authorization
+        response = commit('transactions/refund/for_transaction', post)
+        check_token_response(response, 'transactions/refund/for_transaction', post, options)
+      end
+
+      def void(authorization, options = {})
+        post = {}
+        post[:transaction_id] = authorization
+
+        response = commit('transactions/void', post)
+        check_token_response(response, 'transactions/void', post, options)
+      end
+
+      def verify(credit_card, options = {})
+        MultiResponse.run(:use_first_response) do |r|
+          r.process { authorize(100, credit_card, options) }
+          r.process(:ignore_result) { void(r.authorization, options) }
+        end
+      end
+
+      # The customer_IDs that come from storing cards can be used for auth and purchase transaction types
+      def store(credit_card, options = {})
+        post = {}
+        post[:customer_id] = options[:customer_id] || SecureRandom.hex(12)
+        add_payment(post, credit_card)
+        add_address(post, credit_card, options)
+        response = commit('customer/create', post)
+        check_token_response(response, 'customer/create', post, options)
+      end
+
+      def redact(customer_id)
+        post = {}
+        post[:customer_id] = customer_id
+        response = commit('customer/delete', post)
+        check_token_response(response, 'customer/delete', post, options)
+      end
+
+      def settle
```

Comment:
Since (I think??) this isn't part of the standard AM implementation (but is definitely handy for testing), I'm wondering if we should monkeypatch it into the gateway from the test file rather than in here? 🤔

Reply:
We did this together! 🎉